packages/venia-ui/lib/components/RichContent/PageBuilder/ContentTypes/Tabs/tabs.js

```diff
@@ -0,0 +1,297 @@
+import React, {
+    Children,
+    useRef,
+    useEffect,
+    useCallback,
+    useState
+} from 'react';
+import {
+    Tabs as TabWrapper,
+    TabList,
+    Tab as TabHeader,
+    TabPanel
+} from 'react-tabs';
+import defaultClasses from './tabs.css';
+import { mergeClasses } from '../../../../../classify';
+import { arrayOf, number, oneOf, shape, string } from 'prop-types';
+
+/**
+ * Page Builder Tabs component.
+ *
+ * This component is part of the Page Builder / PWA integration. It can be consumed without Page Builder.
+ *
+ * @typedef Tabs
+ * @kind functional component
+ *
+ * @param {props} props React component props
+ *
+ * @returns {React.Element} A React component that displays a set of Tabs.
+ */
+const Tabs = props => {
+    const classes = mergeClasses(defaultClasses, props.classes);
+    const navigationRef = useRef(null);
+    const [isScrolling, setIsScrolling] = useState(false);
+    const [clientX, setClientX] = useState(0);
+    const [scrollX, setScrollX] = useState(0);
+    const [scrollElement, setScrollElement] = useState(null);
+    const [navWrapperClass, setNavWrapperClass] = useState();
+    const {
+        tabNavigationAlignment = 'left',
+        minHeight,
+        defaultIndex = 0,
+        headers = [],
+        textAlign,
+        border,
+        borderColor,
+        borderWidth,
+        borderRadius,
+        marginTop,
+        marginRight,
+        marginBottom,
+        marginLeft,
+        paddingTop,
+        paddingRight,
+        paddingBottom,
+        paddingLeft,
+        cssClasses = [],
+        children
+    } = props;
+
+    const onMouseDown = useCallback(
+        event => {
+            setIsScrolling(true);
+            setClientX(event.clientX);
+        },
+        [setIsScrolling, setClientX]
+    );
+
+    const onMouseUp = useCallback(() => {
+        setIsScrolling(false);
+    }, [setIsScrolling]);
+
+    const onMouseMove = useCallback(
+        event => {
+            if (isScrolling && scrollElement) {
+                scrollElement.scrollLeft = scrollX + (clientX - event.clientX);
+                setScrollX(scrollElement.scrollLeft);
+                setClientX(event.clientX);
+            }
+        },
+        [isScrolling, scrollElement, scrollX, clientX]
+    );
+
+    useEffect(() => {
+        let navScrollElement;
+        const navigationWrapper = navigationRef.current;
+        const handleScroll = () => {
```

Comment:
I believe this handler can be created with `useCallback()` outside of this effect.

Reply:
Resolved: https://github.com/magento/pwa-studio/pull/1872/commits/dc8fb48bc7c2edba5bd3573da6e9fa28e48adfc1